liblib/CMakeLists.txt

```
@@ -52,6 +52,10 @@ IF(MPI_C_INCLUDE_PATH)
     target_include_directories(netcdf PUBLIC ${MPI_C_INCLUDE_PATH})
 ENDIF(MPI_C_INCLUDE_PATH)
 
+IF(TARGET MPI::MPI_C)
+  target_link_libraries(netcdf MPI::MPI_C)
+ENDIF(TARGET MPI::MPI_C)
```

Linking against the target like this means all the include directories are also added -- which means L51-53 can go.

I'd probably also suggest making this link public:

```suggestion
IF(TARGET MPI::MPI_C)
  target_link_libraries(netcdf PUBLIC MPI::MPI_C)
ENDIF(TARGET MPI::MPI_C)
```

---

Specifying a public link here would require a refactor of more of the CMake codebase as mixing signatures to `target_link_libraries` for a given target is not permitted by CMake.

The netcdf target is linked later in the list file via the "plain-signature" so it cannot be linked with the keyword signature. If there is support for refactoring all of the incidences of the `netcdf` targets link interface, then I'd be happy to do so.

---

Ah yes, good point, forgot about that!